src/workspace-data/Data.js

```diff
@@ -593,37 +593,43 @@ export const WorkspaceData = _.flow(
     }, [workspaceId, wdsApp]);
 
     const loadEntityMetadata = async () => {
-      try {
-        setEntityMetadata(undefined);
-        setEntityMetadataError(false);
-        const entityMetadata = await Ajax(signal).Workspaces.workspace(namespace, name).entityMetadata();
+      if (isGoogleWorkspace) {
```

Comment:
I attempted to check the workspace before even calling these methods [here](https://github.com/DataBiosphere/terra-ui/blob/ea0a78d5c8b84e805bb6316fe87d45148c871ffe/src/workspace-data/Data.js#L635), but couldn't figure out how to make it work.  If that seems a better solution, I welcome advice on how to do so.

Reply:
would something like this work?
```javascript
    const loadMetadata = () => {
      if (isAzureWorkspace) {
        return Promise.all([() => setEntityMetadata({}), () => setSnapshotMetadataError(false), refreshRunningImportJobs(), loadWdsSchema()]);
      }
      return Promise.all([loadEntityMetadata(), loadSnapshotMetadata(), refreshRunningImportJobs(), loadWdsSchema()]);
    };
```
there's probably a much cleaner way to write that, too.

Reply:
I didn't have luck with `return Promise.all`, which I tried some variation on.  I think it should do something other than return it but I don't know what

Reply:
hm, making it a bit more structured:
```javascript
    const azureLoadSnapshotMetadata = async () => {
      setSnapshotMetadataError(false);
    };

    const azureLoadEntityMetadata = async () => {
      setEntityMetadata({});
    };

    const loadMetadata = () =>
      isAzureWorkspace
        ? Promise.all([azureLoadEntityMetadata(), azureLoadSnapshotMetadata(), refreshRunningImportJobs(), loadWdsSchema()])
        : Promise.all([loadEntityMetadata(), loadSnapshotMetadata(), refreshRunningImportJobs(), loadWdsSchema()]);
```
If you think this is the better place to make changes, I'd push on making this work instead of adding more branching code elsewhere